### ISSUE TITLE
Proxy protocol annotation is not working for AWS NLB

### ISSUE THREAD

**Issue by rkubik-hostersi** [kubernetes/kubernetes#101608]
#### What happened:
I'm trying to setup AWS NLB for nginx-ingress-controller via following annotations set:
```
    annotations:
      service.beta.kubernetes.io/aws-load-balancer-type: "nlb"
      service.beta.kubernetes.io/aws-load-balancer-proxy-protocol: "*"
      service.beta.kubernetes.io/aws-load-balancer-backend-protocol: tcp
      service.beta.kubernetes.io/aws-load-balancer-cross-zone-load-balancing-enabled: "true"
      service.beta.kubernetes.io/aws-load-balancer-internal: "true"
```
However, the proxy protocol setting is not being applied to the target groups.
I've verified using CloudWatch Events, that the API call to the AWS during LB creation does not specify any attributes regarding target group proxy protocol setting as below:
```
    "eventName": "CreateTargetGroup",
    "awsRegion": "xxx",
    "sourceIPAddress": "eks.amazonaws.com",
    "userAgent": "eks.amazonaws.com",
    "requestParameters": {
      "unhealthyThresholdCount": 3,
      "healthyThresholdCount": 3,
      "protocol": "TCP",
      "targetType": "instance",
      "healthCheckPort": "traffic-port",
      "vpcId": "vpc-xxx",
      "port": 30832,
      "healthCheckProtocol": "TCP",
      "healthCheckIntervalSeconds": 30,
      "name": "k8s-tools-ingressi-xxx"
    },
```

I've tried to change the configuration via `kubectl edit svc`, but again - empty request was submitted to AWS:
```
    "eventSource": "elasticloadbalancing.amazonaws.com",
    "eventName": "ModifyTargetGroup",
    "awsRegion": "xxx,
    "sourceIPAddress": "eks.amazonaws.com",
    "userAgent": "eks.amazonaws.com",
    "requestParameters": {
      "healthCheckPort": "30617",
      "targetGroupArn": "arn:aws:elasticloadbalancing:xxx:xxx:targetgroup/k8s-tools-ingressi-xxx/xxx"
    },
```

#### What you expected to happen:
Annotation should enable the proxy protocol on target groups.

#### Anything else we need to know?:
AWS EKS, k8s 1.19

**Comment by fawadkhaliq** [comment]
The annotation `service.beta.kubernetes.io/aws-load-balancer-proxy-protocol: "*"` is not supported in the in-tree controller for AWS NLB. The in-tree controller only supports it for the AWS CLB.

We just released AWS load balancer controller v2.2.0 [1], with support for NLB instance mode. It supports setting proxy protocol on AWS NLBs. You can do so in two ways:

1/ use same annotation you're using [2]
```
service.beta.kubernetes.io/aws-load-balancer-proxy-protocol
```

2/ set Target group attributes [3], where you can set any attribute on the Target group. 
```
service.beta.kubernetes.io/aws-load-balancer-target-group-attributes: proxy_protocol_v2.enabled=true
```

fwiw, /2 is the recommended option

[1] https://github.com/kubernetes-sigs/aws-load-balancer-controller/releases/tag/v2.2.0
[2] https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.2/guide/service/annotations/#proxy-protocol-v2
[3] https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.2/guide/service/annotations/#target-group-attributes

**Comment by kishorj** [comment]
@rkubik-hostersi, the proxy protocol annotation will not be supported by the in-tree controller. Please refer to the earlier comments from @fawadkhaliq on the alternative. 

closing this issue.
/close
